extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/StartupBuildSteps.java

```diff
@@ -51,26 +48,15 @@
     private static final Logger LOGGER = Logger.getLogger(StartupBuildSteps.class);
 
     @BuildStep
-    AnnotationsTransformerBuildItem annotationTransformer(CustomScopeAnnotationsBuildItem customScopes) {
-        return new AnnotationsTransformerBuildItem(new AnnotationsTransformer() {
-
-            @Override
-            public boolean appliesTo(org.jboss.jandex.AnnotationTarget.Kind kind) {
-                return kind == org.jboss.jandex.AnnotationTarget.Kind.CLASS;
-            }
-
-            @Override
-            public void transform(TransformationContext context) {
-                if (context.isClass() && !customScopes.isScopeDeclaredOn(context.getTarget().asClass())) {
-                    // Class with no built-in scope annotation but with @Scheduled method
-                    if (Annotations.contains(context.getTarget().asClass().classAnnotations(), STARTUP_NAME)) {
-                        LOGGER.debugf("Found @Startup on a class %s with no scope annotations - adding @ApplicationScoped",
-                                context.getTarget());
-                        context.transform().add(ApplicationScoped.class).done();
-                    }
-                }
-            }
-        });
+    AutoAddScopeBuildItem addScope(CustomScopeAnnotationsBuildItem customScopes) {
+        // Class with no built-in scope annotation but with @Startup method
+        return AutoAddScopeBuildItem.builder()
+                .defaultScope(BuiltinScope.APPLICATION)
+                .match((clazz, annotations, index) -> clazz.classAnnotation(STARTUP_NAME) != null
```

Comment:
You can use `AutoAddScopeBuildItem.Builder.containsAnnotations(DotName...)` instead...

Reply:
@mkouba but this will no longer have the same behaviour. Note that before the annotation transformer only cared about class-level annotations. Your suggestion will make it apply to producers as well. Namely producers without any scope such as [this one in a test](https://github.com/quarkusio/quarkus/blob/master/extensions/arc/deployment/src/test/java/io/quarkus/arc/test/startup/StartupAnnotationTest.java#L157-L162). I'd say that's not intended?

Reply:
Hm, I'm not quite sure. The truth is that we do add `@Dependent` for producers by default. In any case, you can replace the lambda with [AutoAddScopeBuildItem#isAnnotatedWith()](https://github.com/quarkusio/quarkus/blob/master/extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/AutoAddScopeBuildItem.java#L141).

Reply:
`The truth is that we do add @Dependent for producers by default.`

Yeah, and that's pretty much spec based. I don't think we want to change that.

>  In any case, you can replace the lambda with AutoAddScopeBuildItem#isAnnotatedWith().

Uff, I read that class at least twice but never noticed that method. Fridays...